packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -0,0 +1,349 @@
+  end: const Offset(0.0, 0.08),
+);
+
+final Animatable<Offset> _kFullBottomUpTween = Tween<Offset>(
+  begin: const Offset(0.0, 1.0),
+  end: const Offset(0.0, -0.02),
+);
+
+// Offset slightly up when a sheet gets covered by another sheet.
+final Animatable<Offset> _kMidUpTween = Tween<Offset>(
+  begin: Offset.zero,
+  end: const Offset(0.0, -0.005),
+);
+
+// Offset from top of screen to slightly down when covered by a sheet.
+final Animatable<Offset> _kTopDownTween = Tween<Offset>(
+  begin: Offset.zero,
+  end: const Offset(0.0, 0.07),
+);
+
+// Amount the sheet in the background scales down. Found by measuring the width
+// of the sheet in the background and comparing against the screen width. The
+// scale transition will go from a default of 1.0 to 1.0 - _kSheetScaleFactor.
+const double _kSheetScaleFactor = 0.0835;
+
+final Animatable<double> _kScaleTween = Tween<double>(begin: 1.0, end: 1.0 - _kSheetScaleFactor);
+
+/// Docs placeholder
+Future<T?> showCupertinoSheet<T>({
+  required BuildContext context,
+  required WidgetBuilder pageBuilder,
+  bool useRootNavigator = true,
+  bool useNestedNavigation = false,
+}) {
+  final BuildContext topLevelSheetContext = CupertinoSheetController.maybeOf(context)?.topLevelContext ?? context;
+
+  final WidgetBuilder builder = switch (useNestedNavigation) {
+    false => pageBuilder,
+    true => (BuildContext context) {
+      return Navigator(
+        initialRoute: '/',
+        onGenerateRoute: (RouteSettings settings) {
+          return CupertinoPageRoute<void>(
+            builder: (BuildContext context) {
+              return PopScope(
+                canPop: settings.name != '/',
+                onPopInvokedWithResult: (bool didPop, Object? result) {
+                  if (didPop) {
+                    return;
+                  }
+                  Navigator.of(topLevelSheetContext).pop();
+                },
+                child: Semantics(
+                  scopesRoute: true,
+                  explicitChildNodes: true,
+                  child: pageBuilder(context),
+                ),
+              );
+            }
+          );
+        },
+      );
+    }
+  };
+
+  return Navigator.of(topLevelSheetContext, rootNavigator: useRootNavigator).push<T>(CupertinoSheetRoute<T>(
+    pageBuilder: builder,
+  ));
+}
+
+/// Docs placeholder
+class CupertinoSheetTransition extends StatelessWidget {
+  /// Docs placeholder
+  const CupertinoSheetTransition({
+    super.key,
+    required this.primaryRouteAnimation,
+    required this.secondaryRouteAnimation,
+    required this.child,
+  });
+
+  /// Animation
+  final Animation<double> primaryRouteAnimation;
+
+  /// Animation
+  final Animation<double> secondaryRouteAnimation;
+
+  /// The widget below this widget in the tree.
+  final Widget child;
+
+  /// The primary delegated transition. Will slide a non [CupertinoSheetRoute] page down.
+  ///
+  /// If a [CupertinoSheetController] already exists in the stack, then it will
```

Comment:
> in the stack

Actually, if there is a regular route between the existing CupertinoSheetRoute and the new one, then it should it do the normal slide down animation?

Reply:
Right now if a regular route gets pushed to the root navigator after the sheet than the normal delegate logic will happen. I'm not 100% sure what the correct behavior should be. We may want to not play any outgoing transition at all on the sheet.

Reply:
So it won't really be possible to push a normal route on top of a MBS and have it display as a normal route (instead it displays as an additional sheet in the MBS). Can you confirm that we're ok not supporting that? If it's not possible on native iOS or recommended against then this is probably the right call.